src/main/kotlin/com/jahnelgroup/domain/notification/NotificationRepo.kt

```diff
@@ -0,0 +1,13 @@
+package com.jahnelgroup.domain.notification
+
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+
+interface NotificationRepo : JpaRepository<Notification, Long> {
+
+    @Query("select n from Notification n where n.recipient = ?#{ principal?.username } order by n.isRead, n.createdDatetime desc")
```

Comment:
Don't forget your NOLOCKs!!

Reply:
Review comment aside... how does this change the functionality? If you left this line out would it break the application?

Reply:
Oh wait, may have answered my own question here - this query is just for that particular function on line 9, right?

Reply:
correct. I am leaving out the no lock for now as I do not think we want dirty reads on this but down the line someone may want them.